quest_test/test_persistence.py

```diff
@@ -1,14 +1,37 @@
 import asyncio
 from pathlib import Path
+from typing import Callable
+from dotenv import load_dotenv
 
 import pytest
 
+from quest import SqlBlobStorage, DynamoDBBlobStorage
 from src.quest import step
 from src.quest.historian import Historian
-from src.quest.persistence import PersistentHistory, LocalFileSystemBlobStorage
+from src.quest.persistence import PersistentHistory, LocalFileSystemBlobStorage, DynamoDB
 from utils import timeout
+from quest.sql import SQLDatabase, SqlBlobStorage
 
 
+def create_filesystem_storage(path: Path):
+    return LocalFileSystemBlobStorage(path)
+
+def create_sql_storage(path: Path):
+    database = SQLDatabase('sqlite:///:memory:')
+    return SqlBlobStorage(path.name, database.get_session())
+
+def create_dynamodb_storage(path: Path):
+    env_path = Path('.env.integration')
+    load_dotenv(dotenv_path=env_path)
+    dynamodb = DynamoDB()
+    return DynamoDBBlobStorage(path.name, dynamodb.get_table())
+
+storage_funcs = [
+    # create_filesystem_storage,
+    create_sql_storage,
+    # create_dynamodb_storage
+]
```

Comment:
Maybe something like:
```py
storage_funcs = []

if do_filesystem:
    import ...
    def create_filesystem_storage(...): ...
    storage_funcs.append(create_filesystem_storage)

if do_sql:
    import sql, ...
    def create_sql_storage(...): ...
    storage_funcs.append(create_sql_storage)

etc.
```

Reply:
Also, perhaps wrap each blob storage under test in a context:
- the `__enter__` function sets up the storage and returns the blob storage object
- the `__exit__` function cleans up
- the test then uses each context

gist:

```py
class FilesystemStorageContext:
    def __enter__(self):
        self.tmpdir = TemporaryDirectory().__enter__()
        storage = LocalStorage(self.tmpdir)
        return storage

    def __exit__(self, ...):
        self.tmpdir.__exit__(...)

...

storages = [FilesystemStorageContext()]

@parameterize(storages)
def test_stuff(storage_ctx):
    with storage_ctx as storage:
        storage.blah()

```

Reply:
Okay, so is it either the if statements or the contexts? Or do we want to do both?

Reply:
Storages are all contexts